arelle/CntlrCmdLine.py

```diff
@@ -1123,8 +1123,12 @@ def monitorParentProcess():
                         # Anything depending on the ModelXbrl that runs after this may encounter unexpected behavior,
                         # so we'll run it as a final step in the CLI controller flow.
                         ValidateDuplicateFacts.saveDeduplicatedInstance(modelXbrl, deduplicateFactsArg, options.saveDeduplicatedInstance)
+                        if options.keepOpen:
+                            raise RuntimeError(_("Attempted to keep model connection open after saving deduplicated instance. "
```

Comment:
I believe both this and the assert below should be `self.addToLog(..., level=logging.CRITICAL)` and set `success` to false.